### DISCUSSION TITLE
How to create a density plot where density is over all groups?

### DISCUSSION THREAD

**Discussion by stefaneng** [uclahs-cds/public-R-BoutrosLab-plotting-general#104] [Q&A] [answered]
_Originally asked by_ @jeeyunhan 

The default behavior for density plot is to plot independent densities where each group integrates to 1.
How can we create a density plot with multiple groups so that the integral of all groups is 1?

``` r
suppressPackageStartupMessages(library(BoutrosLab.plotting.general));
set.seed(13);
n1 <- 500;
n2 <- 1000;
my.data <- list(
  group1 = rnorm(n1, mean = 2),
  group2 = rnorm(n2, mean = 4)
  );

create.densityplot(
  my.data,
  col = default.colours(2),
  main = 'Independent densities'
  );
```

![](https://i.imgur.com/iURG86J.png)

<sup>Created on 2023-03-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by stefaneng** [comment, marked as answer]
Currently `create.densityplot` does not support this but we can create this plot using `density` function and `create.scatterplot`. We need to compute the density for each group and then scale by the number of data points in each groups.

``` r
suppressPackageStartupMessages(library(BoutrosLab.plotting.general));
set.seed(13);
n1 <- 500;
n2 <- 1000;

my.data <- list(
  group1 = rnorm(n1, mean = 2),
  group2 = rnorm(n2, mean = 4)
  );

grp1.density <- density(my.data$group1);
grp2.density <- density(my.data$group2);

grp1.df <- as.data.frame(grp1.density[c('x', 'y')]);
grp1.df$group <- 1;
# Scale by multiplying by the ratio of: (# of group 1 )/(total count)
grp1.df$scaled.y <- grp1.df$y * n1 / (n1 + n2);
grp2.df <- as.data.frame(grp2.density[c('x', 'y')]);
grp2.df$group <- 2;
# Scale by multiplying by the ratio of: (# of group 2)/(total count)
grp2.df$scaled.y <- grp2.df$y * n2 / (n1 + n2);

combined.df <- rbind(grp1.df, grp2.df);

# Recreate density plot
create.scatterplot(
  y ~ x,
  data = combined.df,
  groups = combined.df$group,
  col = default.colours(2),
  type = 'l'
  );
```

![](https://i.imgur.com/IyxNDlC.png)

``` r

create.scatterplot(
  scaled.y ~ x,
  data = combined.df,
  groups = combined.df$group,
  col = default.colours(2),
  type = 'l'
  );
```

![](https://i.imgur.com/dtYEcPo.png)

<sup>Created on 2023-03-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by pboutros** [comment]
Yup fully agree with @stefaneng on the solution, I was going to offer the same strategy.  I'd also put this as a feature we won't add in the near-term to `create.densityplot()` because it's a bit of a bespoke case, so will tag as `wontfix `for now (but glad the issue/solution are visible here!)
